### PR TITLE
chare: StateAnchor fixes, test refactoring cmt_log/cons

### PR DESCRIPTION
**Pull request by lmoe**
* Changes New/StateAnchor to use the Owner of the AnchorWithRef argument instead of requiring it in the second argument again.
* Fixes all files using the NewStateAnchor 
* Fixes some tests in cons/cmt_log that still uses Iota.go/v3 Stardust types
* Fixes chain deployment in the wasp-cli by using a correct Gas coin

### CODE REVIEW

File: packages/chain/cmt_log/cmt_log_test.go [hunk 1]
```diff
@@ -71,7 +73,7 @@ func testCmtLogBasic(t *testing.T, n, f int) {
 	gpaTC.PrintAllStatusStrings("Initial", t.Logf)
 	//
 	// Provide first alias output. Consensus should be sent now.
-	ao1 := randomAliasOutputWithID(aliasID, governor.Address(), committeeAddress, 1)
+	ao1 := randomAliasOutputWithID(*aliasRef.ObjectID, committeeAddress, 1)
```
Comment (howjmay) [inline packages/chain/cmt_log/cmt_log_test.go]: maybe we can change this to `randomAnchorWithID `

File: packages/chain/cmt_log/cmt_log_test.go [hunk 2]
```diff
@@ -120,21 +122,19 @@ func inputAliasOutputConfirmed(gpaNodes map[gpa.NodeID]gpa.GPA, ao *isc.StateAnc
...
-		AliasID:       aliasID,
-		StateIndex:    stateIndex,
-		StateMetadata: []byte{},
-		Conditions: iotago.UnlockConditions{
-			&iotago.StateControllerAddressUnlockCondition{Address: stateAddress.AsIotagoAddress()},
-			&iotago.GovernorAddressUnlockCondition{Address: governorAddress.AsIotagoAddress()},
-		},
-	}
-	return isc.NewAliasOutputWithID(aliasOutput, outputID)
+func randomAliasOutputWithID(anchorID iotago.ObjectID, stateAddress *cryptolib.Address, stateIndex uint32) *isc.StateAnchor {
```
Comment (howjmay) [inline packages/chain/cmt_log/cmt_log_test.go]: maybe we should refactor `isctest.RandomStateAnchor()` for this behavior

File: packages/chain/cons/cons.go [hunk 3]
```diff
@@ -624,7 +625,7 @@ func (c *consImpl) uponVMOutputReceived(vmResult *vm.VMTaskResult, aggregatedPro
 
 	if vmResult.RotationAddress != nil {
 		// Rotation by the Self-Governed Committee.
-		decidedAnchorRef := vmResult.Task.Anchor.Anchor.ObjectRef
+		decidedAnchorRef := vmResult.Task.Anchor.Anchor().ObjectRef
```
Comment (howjmay) [inline packages/chain/cons/cons.go]: maybe use GetObjectRef()

**Review by howjmay — APPROVED** [review]

File: packages/apilib/deploychain.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ type CreateChainParams struct {
 	Textout              io.Writer
 	Prefix               string
 	StateMetadata        transaction.StateMetadata
+	GasCoinObjectID      *iotago.ObjectID `bcs:"optional"`
```
Comment (dessaya) [inline packages/apilib/deploychain.go]: CreateChainParams is not a BCS-encoded object, is it? So the `bcs:"optional"` looks out of place here.

**Review by dessaya — APPROVED** [review]

**Pull request merged**